### PR TITLE
#136 solr should listen on localhost by default

### LINKED ISSUE #136: limit Solr to localhost
https://daniel.hepper.net/blog/2012/04/restricting-ip-addresses-for-solr-with-jetty/ suggests a way to limit Solr to localhost. Might be a good idea. I'm thinking especially about "2019-07-31: [CVE-2019-0193] Remote Code Execution via DataImportHandler" from https://cwiki.apache.org/confluence/display/solr/SolrSecurity

### PR DESCRIPTION
**Pull request by donsizemore**
closes #136

### CODE REVIEW

File: templates/solr.service.j2 [hunk 1]
```diff
@@ -6,10 +6,11 @@ After = syslog.target network.target remote-fs.target nss-lookup.target
 Type = forking
 User={{ dataverse.solr.user }}
 Group={{ dataverse.solr.group }}
-WorkingDirectory = {{ dataverse.solr.root }}/server/solr/collection1
```
Comment (pdurbin) [inline templates/solr.service.j2]: I'm a little confused why the "collection1" stuff is being removed from "WorkingDirectory". It is related to the "-j" stuff?

File: templates/solr.service.j2 [hunk 2]
```diff
@@ -6,10 +6,11 @@ After = syslog.target network.target remote-fs.target nss-lookup.target
...
-ExecStart = {{ dataverse.solr.root }}/bin/solr start
+WorkingDirectory = {{ dataverse.solr.root }}
+ExecStart = {{ dataverse.solr.root }}/bin/solr start -j "jetty.host={{ dataverse.solr.listen }}"
 ExecStop = {{ dataverse.solr.root }}/bin/solr stop
 Restart=on-failure
+LimitNPROC=65000
```
Comment (pdurbin) [inline templates/solr.service.j2]: Is this "LimitNPROC" addition related to the "-j" stuff?

File: templates/solr.service.j2 (hunk 1 above)
Comment (donsizemore) [reply templates/solr.service.j2]: While I was in the solr template I compared the role's copy against what IQSS recommends in the Dataverse docs, and doctored up the working directory and NPROC while I was in there.

**Comment by pdurbin** [comment]
@kcondon @landreev I haven't tested this at all but I thought I'd get your opinion on if you think we should try to limit Solr to localhost only unless we explicitly want it to communicate over the network. As we discussed at standup, that's how PostgreSQL works. This pull request shows one of two approaches linked from a blog post I put in the issue that this pull request closes.

**Pull request merged**